keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -666,7 +773,1310 @@ ways.  One is that we are dispatching requests to be served rather
+
+### Fair Queuing for Server Requests, with max-min fairness and serial virtual execution
+
+In this quadrant we achieve max-min fairness and a non-empty queue has
+exactly one request executing at a given time in the virtual world.
+
+#### Fair Queuing for Server Requests, with max-min fairness and serial virtual execution, behavior
+
+The following behavior solves the problem.
+
+We imagine a virtual world in which request executions are scheduled
+(queued and executed) differently than in the real world.  In the
+virtual world, requests are generally executed with more or less
+concurrency than in the real world and with infinitesimal
+interleaving.
+
+The virtual world uses the same clock as the real world.
+
+Define the following.
+
+- `len(i,j)` is the execution duration for request `(i,j)` in the real
+  world, including `Y(i,j)`.  At first `len` is only a guess.  When
+  the real execution duration is eventually learned, `len` gets
+  updated with that information.
+
+- `B(i,j)` is the time when request `(i,j)` beings execution in the
+  virtual world.
+
+- `E(i,j)` is the time when request `(i,j)` ends execution in the
+  virtual world (this includes completion of `Y(i,j)`).  This is _not_
+  simply `B` plus `len` because requests generally execute at a
+  different rate in the two worlds.
+
+- `NOS(i,t)` is the Number of Occupied Seats by queue `i` at time `t`;
+  this is `Sum[over j such that B(i,j) <= t < E(i,j)] width(i,j)`.
+
+- `SAQ(t)` is the Set of Active Queues at time `t`: those `i` for
+  which there is a `j` such that `t_arrive(i,j) <= t < E(i,j)`.
+
+- `NEQ(t)` is the number of Non-Empty Queues at time `t`; it is the
+  number of queues in `SAQ(t)`.
+
+At time `t`, queue `i` is requesting `rho(i,t)` concurrency in the
+virtual world.  This is the sum of the widths of the requests that
+have arrived but not yet finished executing.
+
+```
+rho(i,t) = Sum[over j such that t_arrive(i,j) <= t < E(i,j)] width(i,j)
+```
+
+The allocations of concurrency are written as `mu(i,t)` seats for
+queue `i` at time `t`.  These allocations are max-min fair when
+
+```
+mu(i,t) = min(rho(i,t), mu_fair(t))
+```
+
+and `mu_fair(t)` is the minimum non-negative value that solves the
+equation
+
+```
+min(C, Sum[over i] rho(i,t)) = Sum[over i] min(rho(i,t), mu_fair(t))
+```
+
+In English: every queue gets what it wants if that adds up to no more
+than `C`, otherwise queues that request less than the `mu_fair`
+solution get all that they ask for and each of the others gets
+`mu_fair`.
+
+In this design, a queue executes one request at a time in the virtual
+world.  Thus, `NOS(i,t) = width(i,j)` for that relevant `j` whenever
+there is one.
+
+Each non-empty queue divides its allocated concurrency `mu` evenly
+among the seats it occupies in the virtual world, so that the
+aggregate rate work gets done on all the queue's seats is `mu`.
+
+```
+rate(i,t) = if NOS(i,t) > 0 then mu(i,t) / NOS(i,t) else 0
```

Comment:
In fact, this whole subsection is pretty much the same as the section above - with the only difference of mu(i, t) here instead of mu_fair

I'm wondering if we really want to copy 130 lines of text here. Can't we just write what changed compared to the above?